// @xotiny/xb-react-elements@1.1.7/dist/hydrogen/graphql/collection-all-query.js
import { PRODUCT_ITEM_FRAGMENT } from './fragments';
// @see: https://shopify.dev/docs/api/storefront/2025-01/queries/products
export const COLLECTION_ALL_QUERY = `#graphql
  ${PRODUCT_ITEM_FRAGMENT}
  query Products(
    $country: CountryCode
    $language: LanguageCode
    $reverse: Boolean
    $first: Int
    $last: Int
    $startCursor: String
    $endCursor: String
    $sortKey: ProductSortKeys!
  ) @inContext(country: $country, language: $language) {
    products(
      first: $first,
      last: $last,
      before: $startCursor,
      after: $endCursor,
      reverse: $reverse,
      sortKey: $sortKey,
    ) {
      nodes {
        ...ProductItem
      }
      pageInfo {
        hasPreviousPage
        hasNextPage
        endCursor
        startCursor
      }
      filters {
        id
        label
        type
        values {
          id
          label
          count
          input
        }
      }
    }
    collections(first: 100) {
      edges {
        node {
          title
          handle
        }
      }
    }
  }
`;
