// @xotiny/xb-react-elements@1.1.7/dist/hydrogen/graphql/product-query.js
import { MEDIA_FRAGMENT, PRODUCT_VARIANT_FRAGMENT } from './fragments';
// @see: https://shopify.dev/docs/api/storefront/2025-01/queries/product
export const PRODUCT_QUERY = `#graphql
  ${MEDIA_FRAGMENT}
  ${PRODUCT_VARIANT_FRAGMENT}
  query Product(
    $country: CountryCode
    $language: LanguageCode
    $handle: String!
    $selectedOptions: [SelectedOptionInput!]!
  ) @inContext(country: $country, language: $language) {
    product(handle: $handle) {
      id
      title
      vendor
      handle
      descriptionHtml
      description
      encodedVariantExistence
      encodedVariantAvailability
      options {
        name
        optionValues {
          name
          firstSelectableVariant {
          ...ProductVariantFragment
          }
          swatch {
            color
            image {
              previewImage {
                url
              }
            }
          }
        }
      }
      selectedOrFirstAvailableVariant(selectedOptions: $selectedOptions, ignoreUnknownOptions: true, caseInsensitiveMatch: true) {
        ...ProductVariantFragment
      }
      adjacentVariants (selectedOptions: $selectedOptions) {
       ...ProductVariantFragment
      }
      featuredImage {
        id
        altText
        url
        width
        height
      }
      media(first: 100) {
        nodes {
          ...Media
        }
      }
      variants(first: 100) {
        nodes {
          ...ProductVariantFragment
        }
      }
      seo {
        description
        title
      }
      updatedAt
      totalInventory
      collections(first: 1) {
        edges {
          node {
            handle
            title
          }
        }
      }
    }
  }
`;
