// @xotiny/xb-react-elements@1.1.7/dist/elements/shopify/article-date/article-date.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useRouteLoaderData } from '@remix-run/react';
import { useArticle } from "../../../hooks/use-shopify-data";
import { elementAttr } from "../../../utils/template-helpers/element-attr";
import './styles';
const DEFAULT_LOCALE = {
    label: 'United States (USD $)',
    language: 'EN',
    country: 'US',
    currency: 'USD',
};
export const ArticleDate = ({ element }) => {
    const { settings } = element;
    const article = useArticle(element.id);
    const rootData = useRouteLoaderData('root');
    const selectedLocale = rootData?.selectedLocale ?? DEFAULT_LOCALE;
    const { language, country } = selectedLocale;
    if (!article) {
        return null;
    }
    const formattedDate = new Intl.DateTimeFormat(`${language}-${country}`, {
        year: 'numeric',
        month: 'long',
        day: 'numeric',
    }).format(new Date(article.publishedAt));
    return (_jsx("div", { ...elementAttr(element, {
            classNames: ['xb-article-date', settings.static?.class],
        }), children: formattedDate }));
};
