// @xotiny/xb-react-elements@1.1.7/dist/elements/basic/image-comparison-label/image-comparison-label.js
import { jsx as _jsx } from "react/jsx-runtime";
import { elementAttr } from "../../../utils/template-helpers/element-attr";
import { Style } from "../../../utils/template-helpers/style";
export const ImageComparisonLabel = ({ element, parent }) => {
    const { settings } = element;
    const getStyle = () => {
        const labelPosition = Style.create({
            top: 0,
            bottom: 'auto',
        });
        if (parent?.settings.textPosition === 'center') {
            labelPosition.set({
                top: '50%',
                bottom: 'auto',
                transform: 'translateY(-50%)',
            });
        }
        else if (parent?.settings.textPosition === 'bottom') {
            labelPosition.set({
                top: 'auto',
                bottom: 0,
            });
        }
        if (parent?.settings.textEnabled !== 'yes') {
            labelPosition.set({
                display: 'none',
            });
        }
        return labelPosition.get();
    };
    return (_jsx("xo-image-comparison-label", { ...elementAttr(element, {
            classNames: ['xb-image-comparison__icon', settings.static?.class, settings?.modifier?.handle],
            isCustomElement: true,
            style: getStyle(),
        }), "text-binding": "text", children: parent?.settings.textEnabled === 'yes' ? settings.text : '' }));
};
