// @xotiny/xb-react-elements@1.1.7/dist/hydrogen/shared/fake.js
export const header = {
    label: 'Header',
    category: '',
    entities: {
        header: {
            id: 'header',
            elementId: 'section',
            type: 'wrapper',
            label: 'Header',
            parent: '',
            children: ['header_text'],
            settings: {
                sectionWidth: {
                    desktop: {
                        value: 100,
                        unit: '%',
                    },
                },
                zIndex: {
                    value: 1,
                },
                positionType: 'relative',
                positionAbsolute: {
                    desktop: 'center',
                },
            },
        },
        header_text: {
            id: 'header_text',
            elementId: 'paragraph',
            type: 'element',
            label: 'Text',
            parent: 'header',
            settings: {
                text: 'Xo Builder Header',
                dropcapFontSize: {
                    desktop: {
                        unit: 'px',
                        value: 50,
                    },
                },
            },
        },
    },
    order: ['header'],
};
export const footer = {
    label: 'Header',
    category: '',
    entities: {
        footer: {
            id: 'footer',
            elementId: 'section',
            type: 'wrapper',
            label: 'Footer',
            parent: '',
            children: ['footer_text'],
            settings: {
                sectionWidth: {
                    desktop: {
                        value: 100,
                        unit: '%',
                    },
                },
                zIndex: {
                    value: 1,
                },
                positionType: 'relative',
                positionAbsolute: {
                    desktop: 'center',
                },
            },
        },
        footer_text: {
            id: 'footer_text',
            elementId: 'paragraph',
            type: 'element',
            label: 'Text',
            parent: 'footer',
            settings: {
                text: 'Xo Builder Footer',
                dropcapFontSize: {
                    desktop: {
                        unit: 'px',
                        value: 50,
                    },
                },
            },
        },
    },
    order: ['footer'],
};
