// @xotiny/xb-react-elements@1.1.7/dist/hydrogen/shared/setting-default.js
export const settings = {
    pageSettings: {
        layout: {
            'container-width': {
                desktop: {
                    value: 1300,
                    unit: 'px',
                },
                tablet: {
                    value: 992,
                    unit: 'px',
                },
                mobile: {
                    value: 768,
                    unit: 'px',
                },
            },
            'spacing-grid-horizontal': {
                desktop: {
                    value: 30,
                    unit: 'px',
                },
                tablet: {
                    value: 30,
                    unit: 'px',
                },
                mobile: {
                    value: 20,
                    unit: 'px',
                },
            },
            'spacing-grid-vertical': {
                desktop: {
                    value: 30,
                    unit: 'px',
                },
                tablet: {
                    value: 30,
                    unit: 'px',
                },
                mobile: {
                    value: 20,
                    unit: 'px',
                },
            },
        },
        customCode: {
            css: '',
            js: '',
        },
        colors: {
            accent: '#123233',
            title: '#121212',
            text: '#474747',
        },
        additionalSettings: {
            isOverrideGlobalSettings: false,
        },
        background: {
            'background-type': 'standard',
            'background-color': '#ffffff',
            'background-gradient': {
                points: [],
                rotate: 0,
            },
        },
    },
    globalSettings: {
        layout: {
            'container-width': {
                desktop: {
                    value: 1300,
                    unit: 'px',
                },
                tablet: {
                    value: 991,
                    unit: 'px',
                },
                mobile: {
                    value: 767,
                    unit: 'px',
                },
            },
            'spacing-grid-horizontal': {
                desktop: {
                    value: 30,
                    unit: 'px',
                },
                tablet: {
                    value: 30,
                    unit: 'px',
                },
                mobile: {
                    value: 20,
                    unit: 'px',
                },
            },
            'spacing-grid-vertical': {
                desktop: {
                    value: 30,
                    unit: 'px',
                },
                tablet: {
                    value: 30,
                    unit: 'px',
                },
                mobile: {
                    value: 20,
                    unit: 'px',
                },
            },
        },
        customCode: {
            css: '',
            js: '',
        },
        colors: {
            accent: '#123233',
            title: '#121212',
            text: '#474747',
        },
        imageLoader: {
            type: 'loader-1',
            size: {
                value: 40,
            },
            color: 'rgba(0, 0, 0, 1)',
            duration: {
                value: 1300,
                unit: 'ms',
            },
            backgroundColor: 'rgba(0, 0, 0, 0.1)',
        },
        placeholderImage: {
            'tab-type': 'product',
        },
    },
};
