// @xotiny/xb-react-elements@1.1.7/dist/elements/basic/image/image-utils.js
import { Style } from "../../../utils/template-helpers/style";
export function imageUtils({ element }) {
    const { settings } = element;
    const imageWidthDesktop = `${settings.imageWidth.desktop.value}${settings.imageWidth.desktop.unit}`;
    const imageWidthTablet = `${settings.imageWidth.tablet?.value ?? settings.imageWidth.desktop.value}${settings.imageWidth.tablet?.unit ?? settings.imageWidth.desktop.unit}`;
    const imageWidthMobile = `${settings.imageWidth.mobile?.value ?? settings.imageWidth.desktop.value}${settings.imageWidth.mobile?.unit ?? settings.imageWidth.desktop.unit}`;
    const imageHeightDesktop = settings.imageHeight.desktop?.value;
    const imageHeightTablet = settings.imageHeight.tablet?.value ?? imageHeightDesktop;
    const imageHeightMobile = settings.imageHeight.mobile?.value ?? imageHeightTablet;
    function getHeightUnit(screen) {
        if (screen === 'tablet') {
            return settings.imageHeight.tablet ? settings.imageHeight.tablet?.unit : settings.imageHeight.desktop?.unit;
        }
        if (screen === 'mobile') {
            return settings.imageHeight.mobile
                ? settings.imageHeight.mobile?.unit
                : settings.imageHeight.tablet
                    ? settings.imageHeight.tablet?.unit
                    : settings.imageHeight.desktop?.unit;
        }
        return settings.imageHeight.desktop?.unit;
    }
    function getRatio() {
        if (settings.maskImageEnabled === 'no') {
            if (settings.imageRatio === 'square') {
                const ratio = Style.create({
                    '--xb-image-aspect-ratio-desktop': 1,
                    '--xb-image-aspect-ratio-tablet': 1,
                    '--xb-image-aspect-ratio-mobile': 1,
                });
                return ratio.get();
            }
            else if (settings.imageRatio === 'custom') {
                const ratio = Style.create({});
                if (settings.imageHeight.desktop?.unit === '%') {
                    ratio.set({
                        '--xb-image-aspect-ratio-desktop': `100 / ${imageHeightDesktop}`,
                    });
                }
                else {
                    ratio.set({
                        '--xb-image-aspect-ratio-desktop': 'unset',
                        '--xb-image-height-desktop': `${imageHeightDesktop}px`,
                    });
                }
                if (getHeightUnit('tablet') === '%') {
                    ratio.set({
                        '--xb-image-aspect-ratio-tablet': `100 / ${imageHeightTablet}`,
                    });
                }
                else {
                    ratio.set({
                        '--xb-image-aspect-ratio-tablet': 'unset',
                        '--xb-image-height-tablet': `${imageHeightTablet}px`,
                    });
                }
                if (getHeightUnit('mobile') === '%') {
                    ratio.set({
                        '--xb-image-aspect-ratio-mobile': `100 / ${imageHeightMobile}`,
                    });
                }
                else {
                    ratio.set({
                        '--xb-image-aspect-ratio-mobile': 'unset',
                        '--xb-image-height-mobile': `${imageHeightMobile}px`,
                    });
                }
                return ratio.get();
            }
            return settings.image?.width && settings.image?.height
                ? Style.create({
                    '--xb-image-aspect-ratio-desktop': `${settings.image?.width} / ${settings.image?.height}`,
                    '--xb-image-aspect-ratio-tablet': `${settings.image?.width} / ${settings.image?.height}`,
                    '--xb-image-aspect-ratio-mobile': `${settings.image?.width} / ${settings.image?.height}`,
                }).get()
                : {};
        }
        else {
            const ratioMask = [
                { name: 'Mask_Brush_2', width: 559, height: 466 },
                { name: 'Mask_Amorphous_Blob', width: 475, height: 437 },
                { name: 'Mask_Ribbon', width: 467, height: 466 },
                { name: 'Mask_Brutalist_7', width: 468, height: 471 },
                { name: 'Mask_Abstract_4', width: 474, height: 467 },
                { name: 'Mask_Star', width: 471, height: 466 },
                { name: 'MaskImage', width: 833, height: 466 },
                { name: 'Mask_Abstract_3', width: 480, height: 469 },
                { name: 'Mask_Brush_1', width: 479, height: 466 },
                { name: 'Mask_Brush_3', width: 606, height: 466 },
                { name: 'Mask_Brush_Circle', width: 455, height: 466 },
                { name: 'Mask_Abstract_2', width: 510, height: 489 },
                { name: 'Mask_Shield', width: 397, height: 466 },
                { name: 'Mask_Two_Square', width: 462, height: 466 },
                { name: 'Mask_Arch', width: 374, height: 466 },
                { name: 'Mask_Ornament_Square', width: 425, height: 466 },
                { name: 'Mask_Pointer', width: 468, height: 466 },
                { name: 'Mask_Heart', width: 468, height: 466 },
                { name: 'Mask_Bubble_Chat', width: 511, height: 466 },
                { name: 'Mask_Abstract_1', width: 487, height: 466 },
                { name: 'Mask_Brutalist_6', width: 465, height: 471 },
                { name: 'Mask_Half_Circle', width: 438, height: 466 },
                { name: 'Mask_Abstract_5-01', width: 517, height: 497 },
                { name: 'Mask_Abstract_6-01', width: 570, height: 491 },
                { name: 'Mask_Abstract_7-01', width: 486, height: 493 },
            ];
            const currentMaskRatio = ratioMask.find(item => settings.maskImageStyle.includes(item.name));
            return Style.create({
                '--xb-image-aspect-ratio-desktop': `${currentMaskRatio?.width ?? 466} / ${currentMaskRatio?.height ?? 466}`,
                '--xb-image-aspect-ratio-tablet': `${currentMaskRatio?.width ?? 466} / ${currentMaskRatio?.height ?? 466}`,
                '--xb-image-aspect-ratio-mobile': `${currentMaskRatio?.width ?? 466} / ${currentMaskRatio?.height ?? 466}`,
            }).get();
        }
    }
    function getObjectFit() {
        if (settings.maskImageEnabled === 'no' && settings.imageObjectFit === 'contain') {
            return Style.create({
                '--xb-image-object-fit': 'contain',
            }).get();
        }
        return Style.create({
            '--xb-image-object-fit': 'cover',
        }).get();
    }
    function getMaskImageAnimationClass() {
        let maskImageAnimationClass = 'xb-image__mask';
        if (settings.maskImageEnabled === 'yes' && settings.maskImageAnimateEnabled === 'yes') {
            switch (settings.maskImageStyle) {
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Abstract_1.svg?v=1712119221':
                    maskImageAnimationClass = 'xb-image__mask-animate xb-image__mask-animate--1';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Abstract_2.svg?v=1712119222':
                    maskImageAnimationClass = 'xb-image__mask-animate xb-image__mask-animate--2';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Abstract_3.svg?v=1712119222':
                    maskImageAnimationClass = 'xb-image__mask-animate xb-image__mask-animate--3';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Abstract_4.svg?v=1712119221':
                    maskImageAnimationClass = 'xb-image__mask-animate xb-image__mask-animate--4';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Brutalist_4.svg?v=1712119221':
                    maskImageAnimationClass = 'xb-image__mask-animate xb-image__mask-animate--5';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Brutalist_3.svg?v=1712119221':
                    maskImageAnimationClass = 'xb-image__mask-animate xb-image__mask-animate--9';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Brutalist_6.svg?v=1712119221':
                    maskImageAnimationClass = 'xb-image__mask-animate xb-image__mask-animate--10';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Heart.svg?v=1712119222':
                    maskImageAnimationClass = 'xb-image__mask-animate xb-image__mask-animate--7';
                    break;
                default:
                    break;
            }
        }
        return maskImageAnimationClass;
    }
    function getStyle() {
        const style = Style.create(settings.imageSize === 'full'
            ? {
                width: '100%',
                '--xb-image-width-desktop': imageWidthDesktop,
                '--xb-image-width-tablet': imageWidthTablet,
                '--xb-image-width-mobile': imageWidthMobile,
                '--xb-image-caption-background-color': settings.captionBackgroundColor,
            }
            : {
                '--xb-image-width-desktop': imageWidthDesktop,
                '--xb-image-width-tablet': imageWidthTablet,
                '--xb-image-width-mobile': imageWidthMobile,
                '--xb-image-caption-background-color': settings.captionBackgroundColor,
            });
        return style.get();
    }
    const placeholderImage = 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Default_image.png?v=1713954435';
    const actionEnabled = settings.action?.actionType !== 'no';
    const tagName = actionEnabled ? 'a' : 'figure';
    const classLink = actionEnabled ? 'xb-link' : '';
    const classMaskImage = settings.maskImageEnabled === 'yes' ? 'xb-image--mask' : '';
    const classMaskContent = settings.maskImageEnabled === 'yes'
        ? settings.maskImageAnimateEnabled === 'yes'
            ? getMaskImageAnimationClass()
            : 'xb-image__mask'
        : 'xb-image__static';
    return {
        placeholderImage,
        tagName,
        classLink,
        classMaskImage,
        classMaskContent,
        containerStyle: Style.combine(getStyle(), getRatio(), getObjectFit()),
    };
}
