// @xotiny/xb-react-elements@1.1.7/dist/elements/layout/wrapper/wrapper-utils.js
import { classNames } from "../../../utils/class-names";
import { Style } from "../../../utils/template-helpers/style";
export function wrapperUtils({ element }) {
    const { settings } = element;
    const CSS_VARS = {
        WRAPPER_LAYOUT_DESKTOP: '--xb-wld',
        WRAPPER_VERTICAL_SPACING_DESKTOP: '--xb-wvsd',
        WRAPPER_HORIZONTAL_SPACING_DESKTOP: '--xb-whsd',
        WRAPPER_VERTICAL_SPACING_TABLET: '--xb-wvst',
        WRAPPER_VERTICAL_SPACING_MOBILE: '--xb-wvsm',
        WRAPPER_HORIZONTAL_SPACING_TABLET: '--xb-whst',
        WRAPPER_HORIZONTAL_SPACING_MOBILE: '--xb-whsm',
        WRAPPER_WIDTH_DESKTOP: '--xb-wwd',
        WRAPPER_WIDTH_TABLET: '--xb-wwt',
        WRAPPER_WIDTH_MOBILE: '--xb-wwm',
        WRAPPER_VERTICAL_ALIGN_DESKTOP: '--xb-wvad',
        WRAPPER_VERTICAL_ALIGN_TABLET: '--xb-wvat',
        WRAPPER_VERTICAL_ALIGN_MOBILE: '--xb-wvam',
        WRAPPER_HORIZONTAL_ALIGN_DESKTOP: '--xb-whad',
        WRAPPER_HORIZONTAL_ALIGN_TABLET: '--xb-what',
        WRAPPER_HORIZONTAL_ALIGN_MOBILE: '--xb-wham',
        WRAPPER_WRAP_DESKTOP: '--xb-wwrd',
        WRAPPER_WRAP_TABLET: '--xb-wwrt',
        WRAPPER_WRAP_MOBILE: '--xb-wwrm',
        WRAPPER_HORIZONTAL_POSITION_LEFT_DESKTOP: '--xb-whpld',
        WRAPPER_HORIZONTAL_POSITION_RIGHT_DESKTOP: '--xb-whprd',
        WRAPPER_HORIZONTAL_POSITION_LEFT_TABLET: '--xb-whplt',
        WRAPPER_HORIZONTAL_POSITION_RIGHT_TABLET: '--xb-whprt',
        WRAPPER_HORIZONTAL_POSITION_LEFT_MOBILE: '--xb-whplm',
        WRAPPER_HORIZONTAL_POSITION_RIGHT_MOBILE: '--xb-whprm',
        WRAPPER_Z_INDEX: '--xb-wz',
        WRAPPER_POSITION_TYPE: '--xb-wpt',
        WRAPPER_MIN_HEIGHT_DESKTOP: '--xb-wmhd',
        WRAPPER_MIN_HEIGHT_TABLET: '--xb-wmht',
        WRAPPER_MIN_HEIGHT_MOBILE: '--xb-wmhm',
    };
    function getLayoutStyle() {
        const layout = Style.create({
            [CSS_VARS.WRAPPER_LAYOUT_DESKTOP]: settings.layoutStyle,
        });
        return layout.get();
    }
    function getWidth() {
        if (settings.wrapperWidth === 'fullWidth') {
            const width = Style.create({
                [CSS_VARS.WRAPPER_WIDTH_DESKTOP]: '100%',
                [CSS_VARS.WRAPPER_WIDTH_TABLET]: '100%',
                [CSS_VARS.WRAPPER_WIDTH_MOBILE]: '100%',
            });
            return width.get();
        }
        const width = Style.create({
            [CSS_VARS.WRAPPER_WIDTH_DESKTOP]: `${settings.maxWidth.desktop.value}${settings.maxWidth.desktop.unit}`,
        });
        if (settings.maxWidth?.tablet) {
            width.set({
                [CSS_VARS.WRAPPER_WIDTH_TABLET]: `${settings.maxWidth?.tablet.value}${settings.maxWidth?.tablet?.unit}`,
            });
        }
        else {
            width.set({
                [CSS_VARS.WRAPPER_WIDTH_TABLET]: `${settings.maxWidth?.desktop.value}${settings.maxWidth?.desktop?.unit}`,
            });
        }
        if (settings.maxWidth?.mobile) {
            width.set({
                [CSS_VARS.WRAPPER_WIDTH_MOBILE]: `${settings.maxWidth?.mobile.value}${settings.maxWidth?.mobile?.unit}`,
            });
        }
        else {
            width.set({
                [CSS_VARS.WRAPPER_WIDTH_MOBILE]: `${settings.maxWidth?.desktop.value}${settings.maxWidth?.desktop?.unit}`,
            });
        }
        return width.get();
    }
    function getPosition() {
        if (settings.wrapperWidth === 'fullWidth') {
            return {};
        }
        const horizontalPosition = Style.create({});
        if (settings.horizontalPosition.desktop === 'center') {
            horizontalPosition.set({
                [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_LEFT_DESKTOP]: 'auto',
                [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_RIGHT_DESKTOP]: 'auto',
            });
        }
        else if (settings.horizontalPosition.desktop === 'right') {
            horizontalPosition.set({
                [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_LEFT_DESKTOP]: 'auto',
                [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_RIGHT_DESKTOP]: 0,
            });
        }
        if (settings.horizontalPosition?.tablet) {
            if (settings.horizontalPosition.tablet === 'center') {
                horizontalPosition.set({
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_LEFT_TABLET]: 'auto',
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_RIGHT_TABLET]: 'auto',
                });
            }
            else if (settings.horizontalPosition.tablet === 'right') {
                horizontalPosition.set({
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_LEFT_TABLET]: 'auto',
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_RIGHT_TABLET]: 0,
                });
            }
        }
        else {
            if (settings.horizontalPosition.desktop === 'center') {
                horizontalPosition.set({
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_LEFT_TABLET]: 'auto',
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_RIGHT_TABLET]: 'auto',
                });
            }
            else if (settings.horizontalPosition.desktop === 'right') {
                horizontalPosition.set({
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_LEFT_TABLET]: 'auto',
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_RIGHT_TABLET]: 0,
                });
            }
        }
        if (settings.horizontalPosition?.mobile) {
            if (settings.horizontalPosition.mobile === 'center') {
                horizontalPosition.set({
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_LEFT_MOBILE]: 'auto',
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_RIGHT_MOBILE]: 'auto',
                });
            }
            else if (settings.horizontalPosition.mobile === 'right') {
                horizontalPosition.set({
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_LEFT_MOBILE]: 'auto',
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_RIGHT_MOBILE]: 0,
                });
            }
        }
        else {
            if (settings.horizontalPosition.desktop === 'center') {
                horizontalPosition.set({
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_LEFT_MOBILE]: 'auto',
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_RIGHT_MOBILE]: 'auto',
                });
            }
            else if (settings.horizontalPosition.desktop === 'right') {
                horizontalPosition.set({
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_LEFT_MOBILE]: 'auto',
                    [CSS_VARS.WRAPPER_HORIZONTAL_POSITION_RIGHT_MOBILE]: 0,
                });
            }
        }
        return horizontalPosition.get();
    }
    function getHorizontalAlign() {
        const horizontalAlign = Style.create(settings.layoutStyle === 'row'
            ? {
                [CSS_VARS.WRAPPER_HORIZONTAL_ALIGN_DESKTOP]: settings.horizontalAlign.desktop,
            }
            : {
                [CSS_VARS.WRAPPER_VERTICAL_ALIGN_DESKTOP]: settings.horizontalAlign.desktop,
            });
        if (settings.horizontalAlign?.tablet) {
            horizontalAlign.set(settings.layoutStyle === 'row'
                ? {
                    [CSS_VARS.WRAPPER_HORIZONTAL_ALIGN_TABLET]: settings.horizontalAlign?.tablet,
                }
                : {
                    [CSS_VARS.WRAPPER_VERTICAL_ALIGN_TABLET]: settings.horizontalAlign?.tablet,
                });
        }
        else {
            horizontalAlign.set(settings.layoutStyle === 'row'
                ? {
                    [CSS_VARS.WRAPPER_HORIZONTAL_ALIGN_TABLET]: settings.horizontalAlign?.desktop,
                }
                : {
                    [CSS_VARS.WRAPPER_VERTICAL_ALIGN_TABLET]: settings.horizontalAlign?.desktop,
                });
        }
        if (settings.horizontalAlign?.mobile) {
            horizontalAlign.set(settings.layoutStyle === 'row'
                ? {
                    [CSS_VARS.WRAPPER_HORIZONTAL_ALIGN_MOBILE]: settings.horizontalAlign?.mobile,
                }
                : {
                    [CSS_VARS.WRAPPER_VERTICAL_ALIGN_MOBILE]: settings.horizontalAlign?.mobile,
                });
        }
        else {
            horizontalAlign.set(settings.layoutStyle === 'row'
                ? {
                    [CSS_VARS.WRAPPER_HORIZONTAL_ALIGN_MOBILE]: settings.horizontalAlign?.desktop,
                }
                : {
                    [CSS_VARS.WRAPPER_VERTICAL_ALIGN_MOBILE]: settings.horizontalAlign?.desktop,
                });
        }
        return horizontalAlign.get();
    }
    function getVerticalAlign() {
        const verticalAlign = Style.create(settings.layoutStyle !== 'row'
            ? {
                [CSS_VARS.WRAPPER_HORIZONTAL_ALIGN_DESKTOP]: settings.verticalAlign.desktop,
            }
            : {
                [CSS_VARS.WRAPPER_VERTICAL_ALIGN_DESKTOP]: settings.verticalAlign.desktop,
            });
        if (settings.verticalAlign?.tablet) {
            verticalAlign.set(settings.layoutStyle !== 'row'
                ? {
                    [CSS_VARS.WRAPPER_HORIZONTAL_ALIGN_TABLET]: settings.verticalAlign?.tablet,
                }
                : {
                    [CSS_VARS.WRAPPER_VERTICAL_ALIGN_TABLET]: settings.verticalAlign?.tablet,
                });
        }
        else {
            verticalAlign.set(settings.layoutStyle !== 'row'
                ? {
                    [CSS_VARS.WRAPPER_HORIZONTAL_ALIGN_TABLET]: settings.verticalAlign?.desktop,
                }
                : {
                    [CSS_VARS.WRAPPER_VERTICAL_ALIGN_TABLET]: settings.verticalAlign?.desktop,
                });
        }
        if (settings.verticalAlign?.mobile) {
            verticalAlign.set(settings.layoutStyle !== 'row'
                ? {
                    [CSS_VARS.WRAPPER_HORIZONTAL_ALIGN_MOBILE]: settings.verticalAlign?.mobile,
                }
                : {
                    [CSS_VARS.WRAPPER_VERTICAL_ALIGN_MOBILE]: settings.verticalAlign?.mobile,
                });
        }
        else {
            verticalAlign.set(settings.layoutStyle !== 'row'
                ? {
                    [CSS_VARS.WRAPPER_HORIZONTAL_ALIGN_MOBILE]: settings.verticalAlign?.desktop,
                }
                : {
                    [CSS_VARS.WRAPPER_VERTICAL_ALIGN_MOBILE]: settings.verticalAlign?.desktop,
                });
        }
        return verticalAlign.get();
    }
    function getGap() {
        const spacing = Style.create({
            [CSS_VARS.WRAPPER_VERTICAL_SPACING_DESKTOP]: `${settings.verticalSpacing.desktop.value}${settings.verticalSpacing.desktop.unit}`,
            [CSS_VARS.WRAPPER_HORIZONTAL_SPACING_DESKTOP]: `${settings.horizontalSpacing.desktop.value}${settings.horizontalSpacing.desktop.unit}`,
        });
        if (settings.verticalSpacing?.tablet) {
            spacing.set({
                [CSS_VARS.WRAPPER_VERTICAL_SPACING_TABLET]: `${settings.verticalSpacing?.tablet.value}${settings.verticalSpacing?.tablet?.unit}`,
            });
        }
        else {
            spacing.set({
                [CSS_VARS.WRAPPER_VERTICAL_SPACING_TABLET]: `${settings.verticalSpacing?.desktop?.value}${settings.verticalSpacing?.desktop?.unit}`,
            });
        }
        if (settings.verticalSpacing?.mobile) {
            spacing.set({
                [CSS_VARS.WRAPPER_VERTICAL_SPACING_MOBILE]: `${settings.verticalSpacing?.mobile.value}${settings.verticalSpacing?.mobile?.unit}`,
            });
        }
        else {
            spacing.set({
                [CSS_VARS.WRAPPER_VERTICAL_SPACING_MOBILE]: `${settings.verticalSpacing?.desktop?.value}${settings.verticalSpacing?.desktop?.unit}`,
            });
        }
        if (settings.horizontalSpacing?.tablet) {
            spacing.set({
                [CSS_VARS.WRAPPER_HORIZONTAL_SPACING_TABLET]: `${settings.horizontalSpacing?.tablet.value}${settings.horizontalSpacing?.tablet?.unit}`,
            });
        }
        else {
            spacing.set({
                [CSS_VARS.WRAPPER_HORIZONTAL_SPACING_TABLET]: `${settings.horizontalSpacing?.desktop?.value}${settings.horizontalSpacing?.desktop?.unit}`,
            });
        }
        if (settings.horizontalSpacing?.mobile) {
            spacing.set({
                [CSS_VARS.WRAPPER_HORIZONTAL_SPACING_MOBILE]: `${settings.horizontalSpacing?.mobile.value}${settings.horizontalSpacing?.mobile?.unit}`,
            });
        }
        else {
            spacing.set({
                [CSS_VARS.WRAPPER_HORIZONTAL_SPACING_MOBILE]: `${settings.horizontalSpacing?.desktop?.value}${settings.horizontalSpacing?.desktop?.unit}`,
            });
        }
        return spacing.get();
    }
    function getHeight() {
        const minHeight = Style.create({});
        if (settings.heightSelector === 'custom') {
            minHeight.set({
                [CSS_VARS.WRAPPER_MIN_HEIGHT_DESKTOP]: `${settings.minHeight.desktop.value}${settings.minHeight.desktop.unit}`,
            });
            if (settings.minHeight?.tablet) {
                minHeight.set({
                    [CSS_VARS.WRAPPER_MIN_HEIGHT_TABLET]: `${settings.minHeight?.tablet.value}${settings.minHeight?.tablet?.unit}`,
                });
            }
            else {
                minHeight.set({
                    [CSS_VARS.WRAPPER_MIN_HEIGHT_TABLET]: `${settings.minHeight?.desktop?.value}${settings.minHeight?.desktop?.unit}`,
                });
            }
            if (settings.minHeight?.mobile) {
                minHeight.set({
                    [CSS_VARS.WRAPPER_MIN_HEIGHT_MOBILE]: `${settings.minHeight?.mobile.value}${settings.minHeight?.mobile?.unit}`,
                });
            }
            else {
                minHeight.set({
                    [CSS_VARS.WRAPPER_MIN_HEIGHT_MOBILE]: `${settings.minHeight?.desktop?.value}${settings.minHeight?.desktop?.unit}`,
                });
            }
        }
        return minHeight.get();
    }
    function getWrapper() {
        const wrapDesktop = settings.layoutWrap.desktop;
        const wrapTablet = settings.layoutWrap?.tablet ?? wrapDesktop;
        const wrapMobile = settings.layoutWrap?.mobile ?? wrapTablet;
        const wrapper = Style.create({
            [CSS_VARS.WRAPPER_WRAP_DESKTOP]: wrapDesktop,
            [CSS_VARS.WRAPPER_WRAP_TABLET]: wrapTablet,
            [CSS_VARS.WRAPPER_WRAP_MOBILE]: wrapMobile,
        });
        return wrapper.get();
    }
    function getAbsolutePosition() {
        const absoluteClassDesktop = `xb-wrapper--desktop-${settings.positionAbsolute.desktop}`;
        const absoluteClassTablet = settings.positionAbsolute?.tablet
            ? `xb-wrapper--tablet-${settings.positionAbsolute.tablet}`
            : `xb-wrapper--tablet-${settings.positionAbsolute.desktop}`;
        const absoluteClassMobile = settings.positionAbsolute?.mobile
            ? `xb-wrapper--mobile-${settings.positionAbsolute.mobile}`
            : `xb-wrapper--mobile-${settings.positionAbsolute.desktop}`;
        return classNames(absoluteClassDesktop, absoluteClassTablet, absoluteClassMobile);
    }
    function getIndex() {
        const zIndex = Style.create({
            [CSS_VARS.WRAPPER_Z_INDEX]: `${settings.wrapperIndex?.value}`,
        });
        return zIndex.get();
    }
    function getPositionAbsoluteType() {
        const position = Style.create({
            [CSS_VARS.WRAPPER_POSITION_TYPE]: `${settings.positionType}`,
        });
        return position.get();
    }
    const enabledFlexPosition = settings.positionType !== 'static' && settings.positionType !== 'relative';
    const positionClass = enabledFlexPosition ? getAbsolutePosition() : '';
    return {
        style: Style.combine(getLayoutStyle(), getGap(), getWidth(), getHorizontalAlign(), getHeight(), getVerticalAlign(), getWrapper(), getPosition(), getIndex(), getPositionAbsoluteType()),
        positionClass,
    };
}
