// @xotiny/xb-react-elements@1.1.7/dist/elements/shopify/product-image/product-image-utils.js
import { Style } from "../../../utils/template-helpers/style";
export function productImageUtils({ element }) {
    const { settings } = element;
    const imageWidthDesktop = `${settings.imageWidth.desktop.value}${settings.imageWidth.desktop.unit}`;
    const imageWidthTablet = `${settings.imageWidth.tablet?.value ?? settings.imageWidth.desktop.value}${settings.imageWidth.tablet?.unit ?? settings.imageWidth.desktop.unit}`;
    const imageWidthMobile = `${settings.imageWidth.mobile?.value ?? settings.imageWidth.desktop.value}${settings.imageWidth.mobile?.unit ?? settings.imageWidth.desktop.unit}`;
    const imageHeightDesktop = settings.imageHeight.desktop?.value;
    const imageHeightTablet = settings.imageHeight.tablet?.value ?? imageHeightDesktop;
    const imageHeightMobile = settings.imageHeight.mobile?.value ?? imageHeightDesktop;
    const CSS_VARS = {
        PRODUCT_MEDIA_ASPECT_RATIO_DESKTOP: '--xb-product-media-aspect-ratio-desktop',
        PRODUCT_MEDIA_ASPECT_RATIO_TABLET: '--xb-product-media-aspect-ratio-tablet',
        PRODUCT_MEDIA_ASPECT_RATIO_MOBILE: '--xb-product-media-aspect-ratio-mobile',
        PRODUCT_MEDIA_HEIGHT_DESKTOP: '--xb-product-media-height-desktop',
        PRODUCT_MEDIA_HEIGHT_TABLET: '--xb-product-media-height-tablet',
        PRODUCT_MEDIA_HEIGHT_MOBILE: '--xb-product-media-height-mobile',
        PRODUCT_MEDIA_OBJECT_FIT: '--xb-product-media-object-fit',
        PRODUCT_MEDIA_WIDTH_DESKTOP: '--xb-product-media-width-desktop',
        PRODUCT_MEDIA_WIDTH_TABLET: '--xb-product-media-width-tablet',
        PRODUCT_MEDIA_WIDTH_MOBILE: '--xb-product-media-width-mobile',
    };
    function getMaskImageAnimationClass() {
        let maskImageAnimationClass = 'xb-product-image__mask';
        if (settings.maskImageEnabled === 'yes' && settings.maskImageAnimateEnabled === 'yes') {
            switch (settings.maskImageStyle) {
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Abstract_1.svg?v=1712119221':
                    maskImageAnimationClass = 'xb-product-image__mask-animate xb-product-image__mask-animate--1';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Abstract_2.svg?v=1712119222':
                    maskImageAnimationClass = 'xb-product-image__mask-animate xb-product-image__mask-animate--2';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Abstract_3.svg?v=1712119222':
                    maskImageAnimationClass = 'xb-product-image__mask-animate xb-product-image__mask-animate--3';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Abstract_4.svg?v=1712119221':
                    maskImageAnimationClass = 'xb-product-image__mask-animate xb-product-image__mask-animate--4';
                    break;
                case 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Mask_Brutalist_4.svg?v=1712119221':
                    maskImageAnimationClass = 'xb-product-image__mask-animate xb-product-image__mask-animate--5';
                    break;
                default:
                    break;
            }
        }
        return maskImageAnimationClass;
    }
    function getRatio() {
        if (settings.maskImageEnabled !== 'yes') {
            if (settings.imageRatio === 'square') {
                const ratio = Style.create({
                    [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_DESKTOP]: 1,
                    [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_TABLET]: 1,
                    [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_MOBILE]: 1,
                });
                return ratio.get();
            }
            else if (settings.imageRatio === 'custom') {
                const ratio = Style.create({});
                if (settings.imageHeight.desktop?.unit === '%') {
                    ratio.set({
                        [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_DESKTOP]: `100 / ${imageHeightDesktop}`,
                    });
                }
                else {
                    ratio.set({
                        [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_DESKTOP]: 'unset',
                        [CSS_VARS.PRODUCT_MEDIA_HEIGHT_DESKTOP]: `${imageHeightDesktop}px`,
                    });
                }
                if (settings.imageHeight.tablet) {
                    if (settings.imageHeight.tablet?.unit === '%') {
                        ratio.set({
                            [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_TABLET]: `100 / ${imageHeightTablet}`,
                        });
                    }
                    else {
                        ratio.set({
                            [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_TABLET]: 'unset',
                            [CSS_VARS.PRODUCT_MEDIA_HEIGHT_TABLET]: `${imageHeightTablet}px`,
                        });
                    }
                }
                else {
                    if (settings.imageHeight.desktop?.unit === '%') {
                        ratio.set({
                            [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_TABLET]: `100 / ${imageHeightTablet}`,
                        });
                    }
                    else {
                        ratio.set({
                            [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_TABLET]: 'unset',
                            [CSS_VARS.PRODUCT_MEDIA_HEIGHT_TABLET]: `${imageHeightTablet}px`,
                        });
                    }
                }
                if (settings.imageHeight.mobile) {
                    if (settings.imageHeight.mobile?.unit === '%') {
                        ratio.set({
                            [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_MOBILE]: `100 / ${imageHeightMobile}`,
                        });
                    }
                    else {
                        ratio.set({
                            [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_MOBILE]: 'unset',
                            [CSS_VARS.PRODUCT_MEDIA_HEIGHT_MOBILE]: `${imageHeightMobile}px`,
                        });
                    }
                }
                else {
                    if (settings.imageHeight.desktop?.unit === '%') {
                        ratio.set({
                            [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_MOBILE]: `100 / ${imageHeightMobile}`,
                        });
                    }
                    else {
                        ratio.set({
                            [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_MOBILE]: 'unset',
                            [CSS_VARS.PRODUCT_MEDIA_HEIGHT_MOBILE]: `${imageHeightMobile}px`,
                        });
                    }
                }
                return ratio.get();
            }
        }
        else {
            const ratioMask = [
                { name: 'Mask_Brush_2', width: 559, height: 466 },
                { name: 'Mask_Amorphous_Blob', width: 475, height: 437 },
                { name: 'Mask_Ribbon', width: 467, height: 466 },
                { name: 'Mask_Brutalist_7', width: 468, height: 471 },
                { name: 'Mask_Abstract_4', width: 474, height: 467 },
                { name: 'Mask_Star', width: 471, height: 466 },
                { name: 'MaskImage', width: 833, height: 466 },
                { name: 'Mask_Abstract_3', width: 480, height: 469 },
                { name: 'Mask_Brush_1', width: 479, height: 466 },
                { name: 'Mask_Brush_3', width: 606, height: 466 },
                { name: 'Mask_Brush_Circle', width: 455, height: 466 },
                { name: 'Mask_Abstract_2', width: 510, height: 489 },
                { name: 'Mask_Shield', width: 397, height: 466 },
                { name: 'Mask_Two_Square', width: 462, height: 466 },
                { name: 'Mask_Arch', width: 374, height: 466 },
                { name: 'Mask_Ornament_Square', width: 425, height: 466 },
                { name: 'Mask_Pointer', width: 468, height: 466 },
                { name: 'Mask_Heart', width: 468, height: 466 },
                { name: 'Mask_Bubble_Chat', width: 511, height: 466 },
                { name: 'Mask_Abstract_1', width: 487, height: 466 },
                { name: 'Mask_Brutalist_6', width: 465, height: 471 },
                { name: 'Mask_Half_Circle', width: 438, height: 466 },
                { name: 'Mask_Abstract_5-01', width: 517, height: 497 },
                { name: 'Mask_Abstract_6-01', width: 570, height: 491 },
                { name: 'Mask_Abstract_7-01', width: 486, height: 493 },
            ];
            const currentMaskRatio = ratioMask.find(item => settings.maskImageStyle.includes(item.name));
            return Style.create({
                [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_DESKTOP]: `${currentMaskRatio?.width ?? 466} / ${currentMaskRatio?.height ?? 466}`,
                [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_TABLET]: `${currentMaskRatio?.width ?? 466} / ${currentMaskRatio?.height ?? 466}`,
                [CSS_VARS.PRODUCT_MEDIA_ASPECT_RATIO_MOBILE]: `${currentMaskRatio?.width ?? 466} / ${currentMaskRatio?.height ?? 466}`,
            }).get();
        }
    }
    function getObjectFit() {
        const objectFit = Style.create({});
        if (settings.maskImageEnabled !== 'yes') {
            if (settings.imageObjectFit === 'contain') {
                objectFit.set({
                    [CSS_VARS.PRODUCT_MEDIA_OBJECT_FIT]: 'contain',
                });
                return objectFit.get();
            }
            objectFit.set({
                [CSS_VARS.PRODUCT_MEDIA_OBJECT_FIT]: 'cover',
            });
            return objectFit.get();
        }
        return objectFit.get();
    }
    function getStyle() {
        const style = Style.create({
            '--xb-product-media-width-desktop': imageWidthDesktop,
            '--xb-product-media-width-tablet': imageWidthTablet,
            '--xb-product-media-width-mobile': imageWidthMobile,
            '--xb-product-image-mask-svg': `url('${settings.maskImageStyle}')`,
        });
        if (settings.imageSize === 'full') {
            style.set({
                width: '100%',
            });
        }
        return style.get();
    }
    const actionEnabled = settings.linkToProductEnabled === 'yes';
    const tagName = actionEnabled ? 'a' : 'div';
    const classLink = actionEnabled ? 'xb-link' : '';
    const maskImageClass = settings.maskImageEnabled === 'yes'
        ? settings.maskImageAnimateEnabled === 'yes'
            ? getMaskImageAnimationClass()
            : 'xb-product-image__mask'
        : 'xb-product-image__static';
    return {
        tagName,
        maskImageClass,
        classLink,
        getRatio,
        getObjectFit,
        getStyle,
    };
}
