// @xotiny/xb-react-elements@1.1.7/dist/elements/shopify/collection-title/collection-title.js
import { createElement } from 'react';
import { elementAttr } from "../../../utils/template-helpers/element-attr";
import { useCollection } from "../../../hooks/use-shopify-data";
import './styles';
export const CollectionTitle = ({ element }) => {
    const { settings } = element;
    const linkEnabled = settings.linkEnabled === 'yes';
    const titleClass = `xb-heading xb-${settings.htmlTag}`;
    const classLink = linkEnabled ? 'xb-link' : '';
    const linkTag = linkEnabled ? 'a' : 'div';
    const collection = useCollection(element.id);
    if (!collection || !collection?.title) {
        return null;
    }
    return createElement(settings.htmlTag, {
        ...elementAttr(element, {
            classNames: ['xb-collection-title', titleClass, settings.static?.class],
        }),
        children: createElement(linkTag, {
            className: classLink,
            href: `/collections/${collection?.handle}`,
            dangerouslySetInnerHTML: { __html: collection?.title },
        }),
    });
};
