// @xotiny/xb-react-elements@1.1.7/dist/elements/layout/hover-content/hover-content.js
import { jsx as _jsx } from "react/jsx-runtime";
import { classNames } from "../../../utils/class-names";
import { elementAttr } from "../../../utils/template-helpers/element-attr";
import { Style } from "../../../utils/template-helpers/style";
import { cssEasing } from "../../../utils/css-easing";
import './styles';
export const HoverContent = ({ element, children }) => {
    const { settings } = element;
    const elementOmitId = {
        ...element,
        id: '',
    };
    function getLayoutStyle() {
        return { '--xb-wrapper-layout': settings.layoutStyle };
    }
    function getGap() {
        const spacing = Style.create({
            '--xb-wrapper-vertical-spacing-desktop': `${settings.verticalSpacing.desktop.value}${settings.verticalSpacing.desktop.unit}`,
            '--xb-wrapper-horizontal-spacing-desktop': `${settings.horizontalSpacing.desktop.value}${settings.horizontalSpacing.desktop.unit}`,
        });
        if (settings.verticalSpacing?.tablet?.value) {
            spacing.set({ '--xb-wrapper-vertical-spacing-tablet': `${settings.verticalSpacing?.tablet?.value}${settings.verticalSpacing?.tablet?.unit}` });
        }
        if (settings.verticalSpacing?.mobile?.value) {
            spacing.set({ '--xb-wrapper-vertical-spacing-mobile': `${settings.verticalSpacing?.mobile?.value}${settings.verticalSpacing?.mobile?.unit}` });
        }
        if (settings.horizontalSpacing?.tablet?.value) {
            spacing.set({
                '--xb-wrapper-horizontal-spacing-tablet': `${settings.horizontalSpacing?.tablet?.value}${settings.horizontalSpacing?.tablet?.unit}`,
            });
        }
        if (settings.horizontalSpacing?.mobile?.value) {
            spacing.set({
                '--xb-wrapper-horizontal-spacing-mobile': `${settings.horizontalSpacing?.mobile?.value}${settings.horizontalSpacing?.mobile?.unit}`,
            });
        }
        return spacing.get();
    }
    function getAlign() {
        const align = Style.create();
        if (settings.layoutStyle === 'column') {
            align.set({ '--xb-wrapper-vertical-align': settings.horizontalAlign });
        }
        else {
            align.set({ '--xb-wrapper-horizontal-align': settings.horizontalAlign });
        }
        return align.get();
    }
    function getVerticalAlign() {
        const verticalAlign = Style.create();
        if (settings.layoutStyle === 'column') {
            verticalAlign.set({ '--xb-wrapper-horizontal-align': settings.verticalAlign });
        }
        else {
            verticalAlign.set({ '--xb-wrapper-vertical-align': settings.verticalAlign });
        }
        return verticalAlign.get();
    }
    function getWidth() {
        if (settings.widthSelector === 'auto') {
            return {};
        }
        return {
            '--xb-hover-width': `${settings.width.value}${settings.width.unit}`,
        };
    }
    function getHeight() {
        if (settings.heightSelector === 'auto') {
            return {};
        }
        return {
            '--xb-hover-height': `${settings.height.value}${settings.height.unit}`,
        };
    }
    return (_jsx("div", { ...elementAttr(elementOmitId, {
            classNames: ['xb-hover__content', `xb-hover__content--${settings.position}`, settings.cascade ? 'xb-hover__content--cascade' : ''],
            style: {
                '--xb-hover-duration': `${settings.duration.value}${settings.duration.unit}`,
                '--xb-hover-delay': `${settings.delay.value}${settings.delay.unit}`,
                '--xb-hover-easing': cssEasing[settings.easing],
                '--xb-wrapper-wrap': settings.layoutWrap,
                ...getWidth(),
                ...getHeight(),
                ...getLayoutStyle(),
                ...getGap(),
                ...getAlign(),
                ...getVerticalAlign(),
            },
        }), children: _jsx("div", { className: classNames('xb-hover__content-inner', element.id, `xb-hover__content-inner--${settings.effect.desktop}`, `xb-hover__content-inner-tablet--${settings.effect?.tablet ?? settings.effect.desktop}`, `xb-hover__content-inner-mobile--${settings.effect?.mobile ?? settings.effect.desktop}`, settings.cascade ? `xb-hover__content-inner--child-${settings.childEffect}` : ''), children: children }) }));
};
