// @shopify/cli@3.75.4/dist/chunk-57Q6OQG4.js
import {
  basename,
  dirname,
  extname,
  isAbsolute,
  join,
  normalize,
  relative,
  resolve
} from "./chunk-CDBXAE2F.js";
import {
  __commonJS,
  __require,
  __toESM,
  init_cjs_shims
} from "./chunk-PKR7KJ6P.js";

// ../../node_modules/.pnpm/commondir@1.0.1/node_modules/commondir/index.js
var require_commondir = __commonJS({
  "../../node_modules/.pnpm/commondir@1.0.1/node_modules/commondir/index.js"(exports, module) {
    init_cjs_shims();
    var path = __require("path");
    module.exports = function(basedir, relfiles) {
      if (relfiles)
        var files = relfiles.map(function(r) {
          return path.resolve(basedir, r);
        });
      else
        var files = basedir;
      var res = files.slice(1).reduce(function(ps, file) {
        if (!file.match(/^([A-Za-z]:)?\/|\\/))
          throw new Error("relative path without a basedir");
        for (var xs = file.split(/\/+|\\+/), i = 0; ps[i] === xs[i] && i < Math.min(ps.length, xs.length); i++) ;
        return ps.slice(0, i);
      }, files[0].split(/\/+|\\+/));
      return res.length > 1 ? res.join("/") : "/";
    };
  }
});

// ../cli-kit/dist/public/node/path.js
init_cjs_shims();
var import_commondir = __toESM(require_commondir());
import { fileURLToPath } from "url";
function joinPath(...paths) {
  return join(...paths);
}
function normalizePath(path) {
  return normalize(path);
}
function resolvePath(...paths) {
  return resolve(...paths);
}
function relativePath(from, to) {
  return relative(from, to);
}
function isAbsolutePath(path) {
  return isAbsolute(path);
}
function dirname2(path) {
  return dirname(path);
}
function basename2(path, ext) {
  return basename(path, ext);
}
function extname2(path) {
  return extname(path);
}
function relativizePath(path, dir = cwd()) {
  let result = (0, import_commondir.default)([path, dir]), relativePath2 = relative(dir, path), relativeComponents = relativePath2.split("/").filter((component) => component === "..").length;
  return result === "/" || relativePath2 === "" || relativeComponents > 2 ? path : relativePath2;
}
function isSubpath(mainPath, subpath) {
  let relativePath2 = relative(mainPath, subpath);
  return !relativePath2.startsWith("..") && !isAbsolutePath(relativePath2);
}
function moduleDirectory(moduleURL) {
  return dirname2(fileURLToPath(moduleURL));
}
function cwd() {
  return normalize(process.env.INIT_CWD ? process.env.INIT_CWD : process.cwd());
}
function sniffForPath(argv = process.argv) {
  let pathFlagIndex = argv.indexOf("--path");
  if (pathFlagIndex === -1)
    return argv.find((arg) => arg.startsWith("--path="))?.split("=")[1];
  let pathFlag = argv[pathFlagIndex + 1];
  if (!(!pathFlag || pathFlag.startsWith("-")))
    return pathFlag;
}
function sniffForJson(argv = process.argv) {
  return argv.includes("--json") || argv.includes("-j");
}

export {
  joinPath,
  normalizePath,
  resolvePath,
  relativePath,
  isAbsolutePath,
  dirname2 as dirname,
  basename2 as basename,
  extname2 as extname,
  relativizePath,
  isSubpath,
  moduleDirectory,
  cwd,
  sniffForPath,
  sniffForJson
};
