// @xotiny/xb-react-elements@1.1.7/dist/elements/layout/section/section-utils.js
import { classNames } from "../../../utils/class-names";
import { Style } from "../../../utils/template-helpers/style";
export function sectionUtils({ element }) {
    const { settings } = element;
    const gradientAnimatedEnabled = !!settings.styles?.normal?.background?.['background-gradient-animated'];
    const gradientAnimationClass = gradientAnimatedEnabled ? 'xb-section-background-gradient-animated' : '';
    function getWidth() {
        if (settings.containerWidth !== 'boxed') {
            return '';
        }
        const width = Style.create({
            '--xb-container-width-desktop': `${settings.maxWidth.desktop.value}${settings.maxWidth.desktop.unit}`,
        });
        if (settings.maxWidth?.tablet?.value) {
            width.set({
                '--xb-container-width-tablet': `${settings.maxWidth?.tablet?.value}${settings.maxWidth?.tablet?.unit}`,
            });
        }
        if (settings.maxWidth?.mobile?.value) {
            width.set({
                '--xb-container-width-mobile': `${settings.maxWidth?.mobile?.value}${settings.maxWidth?.mobile?.unit}`,
            });
        }
        return width.get();
    }
    function getContainerClass() {
        let container = 'xb-container';
        if (settings.containerWidth === 'boxed') {
            container = 'xb-container';
        }
        else if (settings.containerWidth === 'fullWidth') {
            container = classNames('xb-container', 'xb-full-width');
        }
        return container;
    }
    function getGap() {
        const gap = Style.create({});
        if (settings.containerGapSelector === 'custom') {
            gap.set({
                '--xb-container-gap-desktop': `${settings.containerGap.desktop.value}${settings.containerGap.desktop.unit}`,
            });
            if (settings.containerGap?.tablet) {
                gap.set({
                    '--xb-container-gap-tablet': `${settings.containerGap?.tablet.value}${settings.containerGap?.tablet.unit}`,
                });
            }
            if (settings.containerGap?.mobile) {
                gap.set({
                    '--xb-container-gap-mobile': `${settings.containerGap?.mobile.value}${settings.containerGap?.mobile.unit}`,
                });
            }
        }
        return gap.get();
    }
    function getSectionMinHeight() {
        const minHeight = Style.create({});
        if (settings.heightSelector === 'custom') {
            minHeight.set({
                '--xb-section-min-height-desktop': `${settings.minHeight.desktop.value}${settings.minHeight.desktop.unit}`,
            });
            if (settings.minHeight?.tablet) {
                minHeight.set({
                    '--xb-section-min-height-tablet': `${settings.minHeight?.tablet.value}${settings.minHeight?.tablet.unit}`,
                });
            }
            if (settings.minHeight?.mobile) {
                minHeight.set({
                    '--xb-section-min-height-mobile': `${settings.minHeight?.mobile.value}${settings.minHeight?.mobile.unit}`,
                });
            }
        }
        return minHeight.get();
    }
    function getSectionVerticalAlign() {
        const verticalAlign = Style.create({});
        if (settings.heightSelector === 'custom') {
            verticalAlign.set({
                '--xb-section-align-content-desktop': settings.verticalAlign.desktop,
            });
            if (settings.verticalAlign?.tablet) {
                verticalAlign.set({
                    '--xb-section-align-content-tablet': settings.verticalAlign?.tablet,
                });
            }
            if (settings.verticalAlign?.mobile) {
                verticalAlign.set({
                    '--xb-section-align-content-mobile': settings.verticalAlign?.mobile,
                });
            }
        }
        return verticalAlign.get();
    }
    function getPositionAbsoluteType() {
        const absoluteType = Style.create({
            '--xb-section-position-type': settings.positionType,
        });
        return absoluteType.get();
    }
    function getGradientAnimationDuration() {
        const gradientAnimationDuration = settings.styles?.normal?.background?.['background-gradient-animated-duration']?.value ?? 10;
        const durationStyle = Style.create({
            '--xb-section-background-gradient-animated-duration': `${gradientAnimationDuration}s`,
        });
        return durationStyle.get();
    }
    function getZIndex() {
        const zIndex = Style.create({
            '--xb-section-z-index': settings?.zIndex?.value ?? 0,
        });
        return zIndex.get();
    }
    function getAbsolutePosition() {
        const desktopClass = `xb-section--desktop-${settings.positionAbsolute?.desktop ?? 'relative'}`;
        const tabletClass = settings.positionAbsolute?.tablet ? `xb-section--tablet-${settings.positionAbsolute.tablet}` : desktopClass;
        const mobileClass = settings.positionAbsolute?.mobile ? `xb-section--mobile-${settings.positionAbsolute.mobile}` : tabletClass;
        return classNames(desktopClass, tabletClass, mobileClass);
    }
    const absoluteClass = settings.positionType !== 'static' && settings.positionType !== 'relative' ? getAbsolutePosition() : '';
    function getSectionWidth() {
        const sectionWidth = Style.create({});
        if (settings.positionType === 'static' || settings.positionType === 'relative') {
            sectionWidth.set({
                '--xb-section-width-desktop': '100%',
                '--xb-section-width-tablet': '100%',
                '--xb-section-width-mobile': '100%',
            });
            return sectionWidth.get();
        }
        if (settings.sectionWidth?.desktop?.value) {
            sectionWidth.set({
                '--xb-section-width-desktop': `${settings.sectionWidth?.desktop?.value}${settings.sectionWidth?.desktop?.unit}`,
                '--xb-section-width-tablet': `${settings.sectionWidth?.tablet?.value}${settings.sectionWidth?.tablet?.unit}`,
                '--xb-section-width-mobile': `${settings.sectionWidth?.mobile?.value}${settings.sectionWidth?.mobile?.unit}`,
            });
        }
        return sectionWidth.get();
    }
    function getSectionDividerHeight(height, position) {
        const desktopHeight = height.desktop?.value;
        const tabletHeight = height?.tablet?.value ?? desktopHeight;
        const mobileHeight = height?.mobile?.value ?? tabletHeight;
        const dividerHeight = Style.create(position === 'top'
            ? {
                '--xb-section-divider-top-height-desktop': `${desktopHeight}px`,
                '--xb-section-divider-top-height-tablet': `${tabletHeight}px`,
                '--xb-section-divider-top-height-mobile': `${mobileHeight}px`,
            }
            : {
                '--xb-section-divider-bottom-height-desktop': `${desktopHeight}px`,
                '--xb-section-divider-bottom-height-tablet': `${tabletHeight}px`,
                '--xb-section-divider-bottom-height-mobile': `${mobileHeight}px`,
            });
        return dividerHeight.get();
    }
    return {
        gradientAnimationClass,
        absoluteClass,
        getContainerClass,
        getWidth,
        getGap,
        getSectionDividerHeight,
        containerStyle: Style.combine(getSectionMinHeight(), getSectionVerticalAlign(), getZIndex(), getSectionWidth(), getPositionAbsoluteType(), getGradientAnimationDuration()),
    };
}
