// @xotiny/xb-react-elements@1.1.7/dist/utils/template-helpers/get-css-variable-modifier/index.js
import { objectKeys } from '../../object/object-keys';
export const getPropValue = (prop, value) => {
    if (/\b(font-size|line-height|letter-spacing|opacity|width|height|slider)\b/.test(prop)) {
        if (value.value != null) {
            if (value.unit) {
                return `${prop}: ${value.value}${value.unit};`;
            }
            return `${prop}: ${value.value};`;
        }
        return '';
    }
    if (prop.includes('blur')) {
        return `filter: blur(${value.value}${value.unit});`;
    }
    if (prop.includes('background-image')) {
        return `${prop}: url(${value.url});`;
    }
    if (prop.includes('background-position')) {
        return `${prop}: ${value.x}% ${value.y}%;`;
    }
    if (/\b(padding|margin|border-width|border-radius)\b/.test(prop)) {
        return `${prop}: ${value.top.value}${value.top.unit} ${value.right.value}${value.right.unit} ${value.bottom.value}${value.bottom.unit} ${value.left.value}${value.left.unit};`;
    }
    return `${prop}: ${value};`;
};
export const getCssContentVariableModifier = (cssVariables) => {
    const css = {
        desktop: '',
        tablet: '',
        mobile: '',
    };
    objectKeys(cssVariables).forEach((cssVar) => {
        if (cssVar.includes('device')) {
            if (cssVariables[cssVar].desktop) {
                css.desktop += getPropValue(cssVar, cssVariables[cssVar].desktop);
            }
            if (cssVariables[cssVar].tablet) {
                css.tablet += getPropValue(cssVar, cssVariables[cssVar].tablet);
            }
            if (cssVariables[cssVar].mobile) {
                css.mobile += getPropValue(cssVar, cssVariables[cssVar].mobile);
            }
        }
        else {
            css.desktop += getPropValue(cssVar, cssVariables[cssVar]);
        }
    });
    return css;
};
export const getCssVariableModifier = (className, cssVariables) => {
    let css = '';
    if (className && cssVariables) {
        const cssDevice = getCssContentVariableModifier(cssVariables);
        if (cssDevice.desktop) {
            css += `.${className}{${cssDevice.desktop}}\n`;
        }
        if (cssDevice.tablet) {
            css += `@media screen and (min-width: 768px) and (max-width: 1024px) {
        .${className} {
          ${cssDevice.tablet}
        }
      }`;
        }
        if (cssDevice.mobile) {
            css += `@media screen and (max-width: 767px) {
        .${className} {
          ${cssDevice.mobile}
        }
      }`;
        }
    }
    return css;
};
