// @xotiny/xb-react-elements@1.1.7/dist/elements/shopify/product-quantity/product-quantity.js
import { jsx as _jsx } from "react/jsx-runtime";
import { useProduct } from "../../../hooks/use-shopify-data";
import { elementAttr } from "../../../utils/template-helpers/element-attr";
import './styles';
export const ProductQuantity = ({ element, children }) => {
    const { settings } = element;
    const widthStyle = settings.fullWidthEnabled === 'yes'
        ? { width: '100%', height: `${settings.height.value}px` }
        : {
            minWidth: `${settings.width.value}px`,
            width: `${settings.width.value}px`,
            height: `${settings.height.value}px`,
        };
    const fullWidthClass = settings.fullWidthEnabled === 'yes' ? 'xb-product-quantity--full' : '';
    const { product } = useProduct(element.id);
    if (!product) {
        return null;
    }
    return (_jsx("div", { ...elementAttr(element, {
            classNames: ['xb-product-quantity', fullWidthClass, settings.static?.class],
            style: widthStyle,
        }), children: children }));
};
