// @xotiny/xb-react-elements@1.1.7/dist/config.js
export const CONFIG = {
    API: {
        URL_DEV: 'https://hydrogen.builder.awe7.com/api/v1',
        URL: 'http://h.xobuilder.app/api/v1',
        GLOBAL_SETTINGS: '/hydrogen/settings',
        PAGE_SETTINGS: '/hydrogen/page-settings/',
        HOME: '/hydrogen/pages/home',
        PRODUCT: '/hydrogen/pages/product',
        COLLECTION: '/hydrogen/pages/collection',
        ARTICLE: '/hydrogen/pages/article',
        PAGE: '/hydrogen/pages/page',
    },
    CACHE_MAX_AGE: 30,
    dev: {
        baseURL: 'http://localhost:8082',
        page: '/hydrogen/page',
        settings: '/hydrogen/settings',
    },
    shopifyFieldIds: {
        collection: '$collection',
        collectionList: '$collection_list',
        collectionSource: '$collection_source',
        limit: '$limit',
        product: '$product',
        blog: '$blogPicker',
        article: '$article',
        productSource: '$productSource',
    },
    collectionPage: {
        filterUrlPrefix: 'filter.v.',
        priceRangeDebounce: 500,
    },
};
