// @xotiny/xb-react-elements@1.1.7/dist/constants.js
export const STATIC_FIELD_TYPES = {
    DIVIDER: 'divider',
    ATOMS: 'atoms',
    HIDDEN: 'hidden',
    HEADER: 'header',
    MARK_UP: 'mark_up',
};
export const FIELD_TYPES = {
    TEXT: 'text',
    CHECKBOX: 'checkbox',
    SLIDER: 'slider',
    SELECT: 'select',
    CHILDREN_ID: 'children_id',
    PARENT_ID: 'parent_id',
    ATTRIBUTES: 'attributes',
    VISIBILITY: 'visibility',
    ACTION: 'action',
    STYLES: 'styles',
    MODIFIER: 'modifier',
    CSS_CODE: 'css_code',
    IMAGE_PICKER: 'image_picker',
    IMAGES_PICKER: 'images_picker',
    VIDEO_PICKER: 'video_picker',
    VIDEOS_PICKER: 'videos_picker',
    FONT_PICKER: 'font_picker',
    COLLECTION_PICKER: 'collection_picker',
    COLLECTION_LIST: 'collection_list',
    PRODUCT_PICKER: 'product_picker',
    PRODUCT_LIST: 'product_list',
    PAGE_PICKER: 'page_picker',
    BLOG_PICKER: 'blog_picker',
    ARTICLE_PICKER: 'article_picker',
    COLOR: 'color',
    VIDEO_URL: 'video_url',
    BUTTON_GROUP: 'button_group',
    LIQUID: 'liquid',
    ICON_PICKER: 'icon_picker',
    RICH_TEXT: 'rich_text',
    SWITCH: 'switch',
    GRID_AREA: 'grid_area',
    TABLE: 'table',
    DATE_PICKER: 'date_picker',
    KEYFRAMES: 'keyframes',
    SCREEN_PICKER: 'screen_picker',
    CAROUSEL_ID: 'carousel_id',
    PRESET: 'preset',
    MEGA_MENU: 'mega_menu',
    NAVIGATION: 'navigation',
};
