// @xotiny/xb-react-elements@1.1.7/dist/root-settings-short.js
export const rootSettingsShort = {
    paragraph: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    button: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'stock-counter': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'star-rating': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'shopify-pagination': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-vendor': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-variants-name': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'product-variants-dropdown-value-item': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'product-variants-dropdown-value': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'product-variants-dropdown': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-title': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-swatches-value-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'product-swatches-value': {
        settings: [],
    },
    'product-swatches': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-quantity-plus': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-quantity-minus': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-quantity-input': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-quantity': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-price-wrapper': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-price': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-metafield': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-media-list-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'product-media-list': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
        ],
    },
    'product-media': {
        settings: [],
    },
    'product-main-media': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'product-list-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-list-hover-image': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-list-content': {
        settings: [],
    },
    'product-list': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-description': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-compare-at-price': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-badge-sale': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'product-badge-new': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'pickup-availabilities-modal': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'pickup-availabilities': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'payment-methods-shop-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'payment-methods-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'payment-methods': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'featured-product': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'dynamic-checkout': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'customer-form-label': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'customer-form-input': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'customer-form-field': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'customer-form': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-title': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-metafield': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-list-content': {
        settings: [],
    },
    'collection-list': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-image': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-description': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'badge-wrapper': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'article-title': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'article-tags': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'article-list-content': {
        settings: [],
    },
    'article-list': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'add-to-cart-button': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    wrapper: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    section: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'hover-content': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    hover: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    grid: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    col: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    vitals: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    trustoo: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    stamped: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'shopify-subscriptions': {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    ryviu: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    rivyo: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    omnisend: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    okendo: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    loox: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    lai: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'judge-me-reviews': {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    fera: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    dadao: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'custom-extension': {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    ali: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    aftership: {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'social-link-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'social-link': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'share-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    share: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    qr: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    marquee: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    map: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'list-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    list: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'image-comparison-label': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'image-comparison-icon': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'image-comparison': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    image: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    icon: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'html-video-icon': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'html-video': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    heading: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'external-video': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    divider: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    callout: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    shape: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'masonry-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    masonry: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    instagram: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'hero-slide': {
        settings: [],
    },
    'hero-content': {
        settings: [],
    },
    hero: {
        settings: [],
    },
    'heading-typing': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'gallery-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    gallery: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'custom-code': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'animate-heading': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'accordion-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'accordion-header': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'accordion-content': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    accordion: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'accodion-icon': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'product-quickview-modal': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'product-quickview': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'form-reset-password-label': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'form-reset-password-input': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'form-reset-password': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'form-register-label': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'form-register-input': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'form-register': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'form-contact-input': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'form-contact': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'form-active-account-label': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'form-active-account-input': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'form-active-account': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'feature-article': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    breadcrumb: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'article-image': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'article-date': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'article-content': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'article-author': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    table: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'facebook-page': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    drr: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'timeline-marker': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'timeline-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    timeline: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'tabs-pane': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    tabs: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    progress: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'parallax-scroll': {
        settings: [],
    },
    'parallax-hover': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'lightbox-trigger': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'lightbox-content-header': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'lightbox-content': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    lightbox: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'hotspot-trigger': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'hotspot-item': {
        settings: [],
    },
    'hotspot-content': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    hotspot: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'flip-box-front': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'flip-box-back': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'flip-box': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    counter: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'countdown-time': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'countdown-label': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'countdown-item': {
        settings: [],
    },
    countdown: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'carousel-slide': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'carousel-pagination': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'carousel-navigation-prev': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'carousel-navigation-next': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'carousel-navigation': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'carousel-content': {
        settings: [],
    },
    carousel: {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'hover-cursor-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'hover-cursor': {
        settings: [],
    },
    'sale-badge': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-list-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'article-list-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'featured-article': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'collection-products-item': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-products-content': {
        settings: [],
    },
    'collection-products': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'metafield-countdown': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
        ],
    },
    'page-title': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filtering-vertical': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filtering-horizontal': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filtering-drawer': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-vertical-sidebar': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-sorting': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-refine': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-head-vertical': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-head-other': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-facets-horizontal': {
        settings: [],
    },
    'collection-filter-facets-drawer-trigger': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-facets-drawer-modal': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-facets-drawer': {
        settings: [],
    },
    'collection-filter-content': {
        settings: [],
    },
    'collection-filter': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'shopify-forms': {
        settings: [
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-layout': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-head': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'collection-filter-count': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'hover-show-image': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'hover-show-content': {
        settings: [],
    },
    'hover-show': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'sub-menu': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'mega-menu': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'language-selector': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'currency-selector': {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'menu-hamburger': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    navigation: {
        settings: [],
    },
    'menu-link': {
        settings: [
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
    'menu-item': {
        settings: [],
    },
    menu: {
        settings: [
            {
                id: 'modifier',
                type: 'modifier',
            },
            {
                id: 'navItemStyle',
                type: 'styles',
            },
            {
                id: 'navItemActiveStyle',
                type: 'styles',
            },
            {
                id: 'styles',
                type: 'styles',
            },
            {
                id: 'css_code',
                type: 'css_code',
            },
        ],
    },
};
