// @xotiny/xb-react-elements@1.1.7/dist/elements/advanced/hotspot/hotspot-utils.js
import { useSettings } from "../../../hooks/use-settings";
import { Style } from "../../../utils/template-helpers/style";
export function hotspotUtils({ element }) {
    const { settings } = element;
    const { globalSettings: global, pageSettings: page } = useSettings();
    const tagName = settings.enable_link ? 'a' : 'div';
    const desktopImage = {
        url: settings.image_picker?.desktop?.url ?? '',
        alt: settings.image_picker?.desktop?.alt ?? '',
        width: settings.image_picker?.desktop?.width ?? 0,
        height: settings.image_picker?.desktop?.height ?? 0,
    };
    const tabletImage = {
        url: settings.image_picker?.tablet?.url ?? desktopImage.url,
        alt: settings.image_picker?.tablet?.alt ?? desktopImage.alt,
        width: settings.image_picker?.tablet?.width ?? desktopImage.width,
        height: settings.image_picker?.tablet?.height ?? desktopImage.height,
    };
    const mobileImage = {
        url: settings.image_picker?.mobile?.url ?? desktopImage.url,
        alt: settings.image_picker?.mobile?.alt ?? desktopImage.alt,
        width: settings.image_picker?.mobile?.width ?? desktopImage.width,
        height: settings.image_picker?.mobile?.height ?? desktopImage.height,
    };
    const imageHeightDesktop = settings.imageHeight.desktop?.value;
    const imageHeightTablet = settings.imageHeight.tablet?.value ?? imageHeightDesktop;
    const imageHeightMobile = settings.imageHeight.mobile?.value ?? imageHeightDesktop;
    function getHeightUnit(screen) {
        if (screen === 'tablet') {
            return settings.imageHeight.tablet ? settings.imageHeight.tablet?.unit : settings.imageHeight.desktop?.unit;
        }
        if (screen === 'mobile') {
            return settings.imageHeight.mobile
                ? settings.imageHeight.mobile?.unit
                : settings.imageHeight.tablet
                    ? settings.imageHeight.tablet?.unit
                    : settings.imageHeight.desktop?.unit;
        }
        return settings.imageHeight.desktop?.unit;
    }
    function getRatio() {
        if (settings.imageRatio === 'square') {
            const ratio = Style.create({
                '--xb-image-aspect-ratio-desktop': 1,
                '--xb-image-aspect-ratio-tablet': 1,
                '--xb-image-aspect-ratio-mobile': 1,
            });
            return ratio.get();
        }
        else if (settings.imageRatio === 'custom') {
            const ratio = Style.create({});
            if (settings.imageHeight.desktop?.unit === '%') {
                ratio.set({
                    '--xb-image-aspect-ratio-desktop': `100 / ${imageHeightDesktop}`,
                });
            }
            else {
                ratio.set({
                    '--xb-image-aspect-ratio-desktop': 'unset',
                    '--xb-image-height-desktop': `${imageHeightDesktop}px`,
                });
            }
            if (getHeightUnit('tablet') === '%') {
                ratio.set({
                    '--xb-image-aspect-ratio-tablet': `100 / ${imageHeightTablet}`,
                });
            }
            else {
                ratio.set({
                    '--xb-image-aspect-ratio-tablet': 'unset',
                    '--xb-image-height-tablet': `${imageHeightTablet}px`,
                });
            }
            if (getHeightUnit('mobile') === '%') {
                ratio.set({
                    '--xb-image-aspect-ratio-mobile': `100 / ${imageHeightMobile}`,
                });
            }
            else {
                ratio.set({
                    '--xb-image-aspect-ratio-mobile': 'unset',
                    '--xb-image-height-mobile': `${imageHeightMobile}px`,
                });
            }
            return ratio.get();
        }
        return desktopImage?.width && desktopImage?.height
            ? Style.create({
                '--xb-image-aspect-ratio-desktop': `${desktopImage?.width} / ${desktopImage?.height}`,
                '--xb-image-aspect-ratio-tablet': `${tabletImage?.width ?? desktopImage?.width} / ${tabletImage?.height ?? desktopImage?.height}`,
                '--xb-image-aspect-ratio-mobile': `${mobileImage?.width ?? desktopImage?.width} / ${mobileImage?.height ?? desktopImage?.height}`,
            }).get()
            : '';
    }
    function getObjectFit() {
        if (settings.imageObjectFit === 'contain') {
            return Style.create({
                '--xb-image-object-fit': 'contain',
            }).get();
        }
        return Style.create({
            '--xb-image-object-fit': 'cover',
        }).get();
    }
    return {
        desktopImage,
        tabletImage,
        mobileImage,
        getRatio,
        getObjectFit,
        tagName,
    };
}
