// @shopify/remix-oxygen@2.0.11/dist/production/index.cjs
'use strict';

var serverRuntime = require('@remix-run/server-runtime');

var c=new TextEncoder,l=async(e,t)=>{let r=await y(t,["sign"]),n=c.encode(e),s=await crypto.subtle.sign("HMAC",r,n),o=btoa(String.fromCharCode(...new Uint8Array(s))).replace(/=+$/,"");return e+"."+o},g=async(e,t)=>{let r=e.lastIndexOf("."),n=e.slice(0,r),s=e.slice(r+1),o=await y(t,["verify"]),a=c.encode(n),i=m(atob(s));return await crypto.subtle.verify("HMAC",o,i,a)?n:!1};async function y(e,t){return await crypto.subtle.importKey("raw",c.encode(e),{name:"HMAC",hash:"SHA-256"},!1,t)}function m(e){let t=new Uint8Array(e.length);for(let r=0;r<e.length;r++)t[r]=e.charCodeAt(r);return t}var u=serverRuntime.createCookieFactory({sign:l,unsign:g}),C=serverRuntime.createCookieSessionStorageFactory(u),S=serverRuntime.createSessionStorageFactory(u),k=serverRuntime.createMemorySessionStorageFactory(S);var A=Error.prototype.toString;Error.prototype.toString=function(){return this.stack||A.call(this)};function E({build:e,mode:t,poweredByHeader:r=!0,getLoadContext:n}){let s=serverRuntime.createRequestHandler(e,t);return async o=>{let a=o.method;if((a==="GET"||a==="HEAD")&&o.body)return new Response(`${a} requests cannot have a body`,{status:400});let i=new URL(o.url);if(i.pathname.includes("//"))return new Response(null,{status:301,headers:{location:i.pathname.replace(/\/+/g,"/")}});let p=n?await n(o):void 0,d=await s(o,p);return r&&d.headers.append("powered-by","Shopify, Hydrogen"),d}}function v(e){let t=e.headers;return {requestGroupId:t.get("request-id"),buyerIp:t.get("oxygen-buyer-ip"),cookie:t.get("cookie"),purpose:t.get("purpose")}}

Object.defineProperty(exports, "MaxPartSizeExceededError", {
	enumerable: true,
	get: function () { return serverRuntime.MaxPartSizeExceededError; }
});
Object.defineProperty(exports, "createSession", {
	enumerable: true,
	get: function () { return serverRuntime.createSession; }
});
Object.defineProperty(exports, "data", {
	enumerable: true,
	get: function () { return serverRuntime.data; }
});
Object.defineProperty(exports, "defer", {
	enumerable: true,
	get: function () { return serverRuntime.defer; }
});
Object.defineProperty(exports, "isCookie", {
	enumerable: true,
	get: function () { return serverRuntime.isCookie; }
});
Object.defineProperty(exports, "isSession", {
	enumerable: true,
	get: function () { return serverRuntime.isSession; }
});
Object.defineProperty(exports, "json", {
	enumerable: true,
	get: function () { return serverRuntime.json; }
});
Object.defineProperty(exports, "redirect", {
	enumerable: true,
	get: function () { return serverRuntime.redirect; }
});
Object.defineProperty(exports, "redirectDocument", {
	enumerable: true,
	get: function () { return serverRuntime.redirectDocument; }
});
exports.createCookie = u;
exports.createCookieSessionStorage = C;
exports.createMemorySessionStorage = k;
exports.createRequestHandler = E;
exports.createSessionStorage = S;
exports.getStorefrontHeaders = v;
