// @xotiny/xb-react-elements@1.1.7/dist/elements/shopify/customer-form-input/customer-form-input.js
import { jsx as _jsx } from "react/jsx-runtime";
import { elementAttr } from "../../../utils/template-helpers/element-attr";
import { Style } from "../../../utils/template-helpers/style";
import './styles';
export const CustomerFormInput = ({ element, parent }) => {
    const { settings } = element;
    function getStyle() {
        const style = Style.create({
            '--xb-input-placeholder-color': settings.placeholderColor,
        });
        return style.get();
    }
    const inputType = settings.type === 'email' ? 'email' : 'text';
    const inputAttrs = settings.type === 'email'
        ? {
            autoCorrect: 'off',
            autoCapitalize: 'off',
            autoComplete: 'email',
        }
        : {};
    const modifierClass = parent?.settings?.modifier?.handle ? `${parent?.settings?.modifier?.handle}__input` : '';
    return (_jsx("input", { ...elementAttr(element, {
            classNames: ['xb-reset', 'xb-form-input', 'xb-customer-form__input', settings.static?.class, modifierClass],
            style: getStyle(),
        }), ...inputAttrs, type: inputType, name: `contact[${settings.type}]`, placeholder: settings.placeholder }));
};
