// @xotiny/xb-react-elements@1.1.7/dist/elements/shopify/collection-image/collection-image-utils.js
import { Style } from "../../../utils/template-helpers/style";
export function collectionImageUtils({ element }) {
    const { settings } = element;
    const imageWidthDesktop = `${settings.imageWidth.desktop.value}${settings.imageWidth.desktop.unit}`;
    const imageWidthTablet = `${settings.imageWidth.tablet?.value ?? settings.imageWidth.desktop.value}${settings.imageWidth.tablet?.unit ?? settings.imageWidth.desktop.unit}`;
    const imageWidthMobile = `${settings.imageWidth.mobile?.value ?? settings.imageWidth.desktop.value}${settings.imageWidth.mobile?.unit ?? settings.imageWidth.desktop.unit}`;
    const imageHeightDesktop = settings.imageHeight.desktop?.value;
    const imageHeightTablet = settings.imageHeight.tablet?.value ?? imageHeightDesktop;
    const imageHeightMobile = settings.imageHeight.mobile?.value ?? imageHeightTablet;
    function getHeightUnit(screen) {
        if (screen === 'tablet') {
            return settings.imageHeight.tablet ? settings.imageHeight.tablet?.unit : settings.imageHeight.desktop?.unit;
        }
        if (screen === 'mobile') {
            return settings.imageHeight.mobile
                ? settings.imageHeight.mobile?.unit
                : settings.imageHeight.tablet
                    ? settings.imageHeight.tablet?.unit
                    : settings.imageHeight.desktop?.unit;
        }
        return settings.imageHeight.desktop?.unit;
    }
    function getRatio() {
        if (settings.imageRatio === 'square') {
            const ratio = Style.create({
                '--xb-collection-image-aspect-ratio-desktop': 1,
                '--xb-collection-image-aspect-ratio-tablet': 1,
                '--xb-collection-image-aspect-ratio-mobile': 1,
            });
            return ratio.get();
        }
        else if (settings.imageRatio === 'custom') {
            const ratio = Style.create({});
            if (settings.imageHeight.desktop?.unit === '%') {
                ratio.set({
                    '--xb-collection-image-aspect-ratio-desktop': `100 / ${imageHeightDesktop}`,
                });
            }
            else {
                ratio.set({
                    '--xb-collection-image-aspect-ratio-desktop': 'unset',
                    '--xb-collection-image-height-desktop': `${imageHeightDesktop}px`,
                });
            }
            if (getHeightUnit('tablet') === '%') {
                ratio.set({
                    '--xb-collection-image-aspect-ratio-tablet': `100 / ${imageHeightTablet}`,
                });
            }
            else {
                ratio.set({
                    '--xb-collection-image-aspect-ratio-tablet': 'unset',
                    '--xb-collection-image-height-tablet': `${imageHeightTablet}px`,
                });
            }
            if (getHeightUnit('mobile') === '%') {
                ratio.set({
                    '--xb-collection-image-aspect-ratio-mobile': `100 / ${imageHeightMobile}`,
                });
            }
            else {
                ratio.set({
                    '--xb-collection-image-aspect-ratio-mobile': 'unset',
                    '--xb-collection-image-height-mobile': `${imageHeightMobile}px`,
                });
            }
            return ratio.get();
        }
    }
    function getObjectFit() {
        if (settings.imageObjectFit === 'contain') {
            const objectFit = Style.create({
                '--xb-collection-image-object-fit-fit': 'contain',
            });
            return objectFit.get();
        }
        const objectFit = Style.create({
            '--xb-collection-image-object-fit-fit': 'cover',
        });
        return objectFit.get();
    }
    function getStyle() {
        const style = Style.create({
            '--xb-image-width-desktop': imageWidthDesktop,
            '--xb-image-width-tablet': imageWidthTablet,
            '--xb-image-width-mobile': imageWidthMobile,
        });
        if (settings.imageSize === 'full') {
            style.set({
                width: '100%',
            });
        }
        return style.get();
    }
    const actionEnabled = settings.linkToCollectionEnabled === 'yes';
    const tagName = actionEnabled ? 'a' : 'div';
    const classLink = actionEnabled ? 'xb-link' : '';
    const defaultPlaceholder = 'https://cdn.shopify.com/s/files/1/0677/7900/2622/files/Collection-placeholder.svg?v=1717754462';
    return {
        tagName,
        classLink,
        getRatio,
        getObjectFit,
        getStyle,
        defaultPlaceholder,
    };
}
