// @xotiny/xb-react-elements@1.1.7/dist/utils/template-helpers/get-animate-attr/index.js
export function getAnimateAttr(tagName, options = {}) {
    const privateOptions = {
        cascade: true,
        enabled: options?.animationEnabled ?? false,
        type: options?.animationType ?? 'infinite',
        duration: options?.animateDuration?.value ?? undefined,
        effect: options?.animateEffect ?? undefined,
        infiniteEffect: options?.animateInfiniteEffect ?? undefined,
        strength: options?.animateStrength?.value ?? undefined,
        ...options,
    };
    if (privateOptions.type === 'infinite') {
        return `xb-animation-infinite="${privateOptions.infiniteEffect}" style="--xb-animation-infinite-duration: ${privateOptions.duration}ms"`;
    }
    let result = `is="xo-animate-${tagName}"`;
    if (!privateOptions.enabled) {
        result += ' xo-disabled';
        return result;
    }
    if (privateOptions.cascade) {
        result += ' xo-cascade';
    }
    if (privateOptions.duration) {
        result += ` xo-duration="${privateOptions.duration}"`;
    }
    if (privateOptions.effect) {
        result += ` xo-type="${privateOptions.effect}"`;
    }
    if (privateOptions.strength) {
        result += ` xo-strength="${privateOptions.strength}"`;
    }
    return result;
}
